packages/gestalt/src/DropdownItem.js

```diff
@@ -0,0 +1,93 @@
+// @flow strict
+import React, { type Node } from 'react';
+import PropTypes from 'prop-types';
+import MenuOption from './MenuOption.js';
+import DropdownContext from './DropdownContextProvider.js';
+
+type OptionObject = {|
+  label: string,
+  value: string,
+  subtext?: string,
+|};
+
+type OptionProps = {|
+  badgeText?: string,
+  children?: Node,
+  isExternal?: boolean,
+  /* the option with value and label */
+  option: OptionObject,
+  selected?: OptionObject | Array<OptionObject> | null,
+  handleSelect?: ({|
+    item: OptionObject,
+    event: SyntheticFocusEvent<HTMLInputElement>,
+  |}) => void,
+  url?: string,
+  // Private prop, not listed in props table
```

Comment:
Since comments easily go stale or lose the lines they're commenting on, consider splitting this out to PublicProps and PrivateProps and then spreading those together to compose Props